Api/Controllers/RestaurantTagsController.cs

```diff
@@ -18,4 +22,39 @@ public async Task<ActionResult<List<string>>> GetAll()
     {
         return Ok(await context.RestaurantTags.Select(rt => rt.Name).ToListAsync());
     }
+
+    /// <summary>
+    /// Gets all restaurants with the given tag
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet]
+    [Route("{tag}/restaurants")]
+    public async Task<ActionResult<List<RestaurantSummaryVM>>> GetRestaurantsWithTag(string tag)
+    {
+        var result = await context.Restaurants
+            .Where(r => r.Tags != null && r.Tags.Select(t => t.Name).Contains(tag))
```

Comment:
lepiej byłoby dodać kolekcję nawigacyjną dla restauracji do klasy `RestaurantTag`, bo to upraszcza SQL i można też zwrócić 404 jeśli takiego taga nie znaleziono